packages/block-editor/src/components/iframe/index.js

```diff
@@ -273,13 +273,16 @@ function Iframe( {
 				src={ src }
 				title={ __( 'Editor canvas' ) }
 				onKeyDown={ ( event ) => {
+					if ( props.onKeyDown ) {
+						props.onKeyDown( event );
+					}
 					// If the event originates from inside the iframe, it means
 					// it bubbled through the portal, but only with React
 					// events. We need to to bubble native events as well,
 					// though by doing so we also trigger another React event,
 					// so we need to stop the propagation of this event to avoid
 					// duplication.
-					if (
+					else if (
```

Comment:
This doesn't make any sense. When you have multiple keydown handlers, all of them should execute

Reply:
The only time the `if` statement above it runs is when a keydown listener is passed to the iframe, which is only in the `view` mode of the site editor. The shortcuts shouldn't be enabled at that point since we're not within the editor.

Reply:
Ok, but that's not clear from reading this. Also there might be other iframe instances that pass onKeydown and do want shortcuts disabled. Using onKeydown to disable/enable shortcuts seems a bit strange, imo that should be done through a separate prop. Maybe reuse `isPreviewMode`?

Reply:
I'm not sure how all of that should be structured. The only thing important to me in this PR was fixing a bug where you couldn't use the keyboard to enter into Edit mode from View mode. [This onKeydown was not allowed to run](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/block-editor/editor-canvas.js#L64-L71).

Very open to fixing this another way. I wasn't sold on this solution.

Reply:
No, this is fine, all I'm saying is that the `else` shouldn't have been added?

Reply:
Oh - That's fine with me: https://github.com/WordPress/gutenberg/pull/59474/